### PR TITLE
Setup scripts improvements (no more Java 8)

### PR DESCRIPTION
**Pull request by julianoes**
This updates the setup scripts quite a bit, mostly because we no longer require Java 8 (since https://github.com/PX4/jMAVSim/pull/119). For the rest of the change, check the commits please.

### CODE REVIEW

File: Tools/setup/OSX.sh [hunk 1]
```diff
@@ -1,64 +1,4 @@
...
-if [[ $INSTALL_SIM == "--sim-tools" ]]; then
-  if brew ls --versions px4-sim > /dev/null; then
-    brew install px4-sim
-  elif [[ $REINSTALL_FORMULAS == "--reinstall" ]]; then
-    brew reinstall px4-sim
-  fi
-fi
-
-echo "All set! PX4 toolchain installed!"
+echo "The script is now called 'macos.sh'."
```
Comment (dagar) [inline Tools/setup/OSX.sh:3]: I'd rather we just delete it.

File: Tools/setup/arch.sh [hunk 2]
```diff
@@ -121,7 +121,8 @@ if [[ $INSTALL_SIM == "true" ]]; then
 	# java (jmavsim or fastrtps)
 	sudo pacman -S --noconfirm --needed \
 		ant \
-		jdk8-openjdk \
+		jre-openjdk \
```
Comment (MaEtUgR) [inline Tools/setup/arch.sh:124]:
```suggestion
```
jre is part of jdk so you don't need to install it separately.

**Comment by hamishwillee** [comment]
Looks good to me,  except the macos build failed due to missing ccache. And of course I don't have merge rights.

**Comment by julianoes** [comment]
>  except the macos build failed due to missing ccache

What? It should not fail without ccache. ccache is completely optional.

**Comment by hamishwillee** [comment]
Don't shoot the messenger - shoot CI

**Comment by julianoes** [comment]
@hamishwillee sorry, let me check :)

**Comment by julianoes** [comment]
@hamishwillee it will come here: https://github.com/PX4/homebrew-px4/pull/58

**Comment by julianoes** [comment]
I'd say let's merge this.

**Pull request merged**